index.d.ts

```diff
@@ -173,13 +172,13 @@ declare namespace Moleculer {
 		broker: ServiceBroker;
 		logger: LoggerInstance;
 		actions?: ServiceActions;
-		Promise: typeof Bluebird;
+		Promise: typeof Promise;
```

Comment:
I think it would be better to use `PromiseConstructorLike` here since typing it as `typeof Promise` (which is the same as `PromiseConstructor`) precludes it being a `Bluebird` style promise.

Reply:
I changed `Promise` to `PromiseConstructorLike`, but it gives the error below. I had that error before and kept ` Promise` to satisfy the type guard:

```index.d.ts:175:19 - error TS2693: 'PromiseConstructorLike' only refers to a type, but is being used as a value here.
175  	Promise: typeof PromiseConstructorLike;
     	                ~~~~~~~~~~~~~~~~~~~~~~
index.d.ts:380:19 - error TS2693: 'PromiseConstructorLike' only refers to a type, but is being used as a value here.
380  	Promise: typeof PromiseConstructorLike;
     	                ~~~~~~~~~~~~~~~~~~~~~~
```
Do we need type guard for Promise here? What if we removed them?

```typescript
class ServiceBroker {
		constructor(options?: BrokerOptions);

		Promise: typeof PromiseConstructorLike;
		// ...
}
```

```typescript
class Service implements ServiceSchema {
		constructor(broker: ServiceBroker, schema?: ServiceSchema);

		protected parseServiceSchema(schema: ServiceSchema): void;

		name: string;
		version?: string | number;
		settings: ServiceSettingSchema;
		metadata: GenericObject;
		dependencies: string | GenericObject | Array<string> | Array<GenericObject>;
		schema: ServiceSchema;
		broker: ServiceBroker;
		logger: LoggerInstance;
		actions?: ServiceActions;
		Promise: typeof PromiseConstructorLike;

		waitForServices(serviceNames: string | Array<string> | Array<GenericObject>, timeout?: number, interval?: number): PromiseLike<void>;

		_init(): void;
		_start(): PromiseLike<void>;
		_stop(): PromiseLike<void>;
		[name: string]: any;
	}
```